pkg/prometheus/operator.go

```diff
@@ -1076,9 +1076,9 @@ func (c *Operator) sync(key string) error {
 		return err
 	}
 
-	// If no service monitor selectors are configured, the user wants to
-	// manage configuration themselves.
-	if p.Spec.ServiceMonitorSelector != nil {
```

Comment:
Would this mean that podSelectors suffer from the same issue if they are the only configuration set?

Reply:
Yeah. After getting some feedback on the original ticket, we decided to remove this gate entirely.